### PR TITLE
Add support for parentLocales

### PR DESCRIPTION
**Pull request by natemueller**
Use the parentLocales section to figure out the correct fallback locales.  The new test passes when I unpack a modern CLDR in vendor/cldr.  I wasn't sure what the proper test procedure is but that worked so I went with it.  Let me know if I should change anything.

### CODE REVIEW

File: lib/cldr/export/data/parent_locales.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+require 'nokogiri'
+
+module Cldr
+  module Export
+    module Data
+      class ParentLocales < Hash
+        def initialize
+          path = "#{Cldr::Export::Data.dir}/supplemental/supplementalData.xml"
```
Comment (camertron) [inline lib/cldr/export/data/parent_locales.rb]: Maybe go with a `File.join` here?

File: lib/cldr/export.rb [hunk 2]
```diff
@@ -119,11 +119,17 @@ def shared_data(component, options = {})
       end
 
       def locales(locale, component, options)
-        locale = locale.to_s.gsub('_', '-').to_sym
+        to_i18n = lambda {|l| l.to_s.gsub('_', '-').to_sym }
```
Comment (camertron) [inline lib/cldr/export.rb]: This is cool, but why not make it a method instead of a lambda?

File: lib/cldr/export/data/parent_locales.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+          doc = File.open(path) { |file| Nokogiri::XML(file) }
+
+          currency_digits_and_rounding = doc.xpath('//parentLocales/parentLocale').each do |node|
```
Comment (camertron) [inline lib/cldr/export/data/parent_locales.rb]: Copy/paste artifact? 😛

**Review by camertron — CHANGES_REQUESTED** [review]
A few comments, but nothing major. Looks good overall, thanks for the PR.

That said, would you mind targeting the original repo (svenfuchs/ruby-cldr) instead of my fork? That's still the official ruby-cldr repo.

**Comment by natemueller** [comment]
Sure, easy changes.  I targeted this one because it's what the twitter-cldr-rb's Gemfile uses.  If I submit these to svenfuchs should I also change switch twitter-cldr-rb to use that version?

File: lib/cldr/download.rb [hunk 4]
```diff
@@ -5,7 +5,7 @@
 module Cldr
   class << self
     def download(source = nil, target = nil)
-      source ||= 'http://unicode.org/Public/cldr/24/core.zip'
+      source ||= 'http://unicode.org/Public/cldr/26/core.zip'
```
Comment (camertron) [inline lib/cldr/download.rb]: This is fine, just FYI that twitter-cldr-rb targets CLDR v29.

File: lib/cldr/export.rb [hunk 5]
```diff
@@ -119,11 +119,25 @@ def shared_data(component, options = {})
...
+        from_i18n = lambda {|l| l.to_s.gsub('-', '_') }
+        locale = to_i18n[locale]
 
         locales = if options[:merge]
-          # passing locale itself into constructor as a default fallback to avoid falling back to :en locale
-          I18n::Locale::Fallbacks.new([locale])[locale]
+          defined_parents = Cldr::Export::Data::ParentLocales.new
+
+          ancestory = [locale]
```
Comment (camertron) [inline lib/cldr/export.rb]: Spelling: "ancestry"

**Review by camertron — APPROVED** [review]

**Comment by camertron** [comment]
Just a couple of other minor comments. Not sure why the fields stuff is showing up in this diff either. Will investigate.

**Comment by natemueller** [comment]
I rebased my change onto svenfuchs/ruby-cldr master, so it updated the PR to include one of your earlier changes.  Thought I closed this one out.  Sorry about the confusion.

**Comment by camertron** [comment]
Got it, thanks. I'll merge the upstream changes when the svenfuchs PR gets merged.

**Pull request closed without merging**